src/bigcommerce.js

```diff
@@ -147,17 +182,52 @@
   }
 
   /**
-   * Shopgate iOS App polling
+   * Enabling ShopgateApp events by injecting a libshopgate meta tag.
+   * Needed early (in the page loading phase) in order for iOS app to work as we expect it to.
+   */
+  function enableShopgateAppEvents() {
+    // Check if insertion is needed
+    var libshopgate = 'libshopgate';
+    if (document.getElementById(libshopgate)) {
+      return;
+    }
+
+    // Insert libshopgate as meta tag, to tell the Shopgate app to send events
+    // Not using a script tag to avoid "src unavailable" errors in the browsers console
+    var metaTag = document.createElement('meta');
+    metaTag.setAttribute('id', libshopgate);
+    // Add a "src" property (not an attribute, because of the iOS app not receiving it otherwise)
+    metaTag.src = libshopgate;
+    document.getElementsByTagName('head').item(0).appendChild(metaTag);
+  }
+
+  /**
+   * Send all given commands to the app
+   * @param {Array} commands
+   */
+  function sendCommandsToApp(commands) {
+    if ('dispatchCommandsForVersion' in SGJavascriptBridge) {
+      SGJavascriptBridge.dispatchCommandsForVersion(commands, '9.0');
+    } else {
+      SGJavascriptBridge.dispatchCommandsStringForVersion(JSON.stringify(commands), '9.0');
+    }
+  }
+
+  /**
+   * Shopgate App code execution
+   * @param intervalInMilliseconds
+   * @param maximumIntervalTimeInMilliseconds
+   * @param executeCallback
    */
   function shopgateExecuteWithRetry(
-    intervalInMiliseconds,
-    maximumIntervallTimeInMiliseconds,
+    intervalInMilliseconds,
+    maximumIntervalTimeInMilliseconds,
     executeCallback
   ) {
-    const startTimestampInMiliseconds = Date.now();
+    var startTimestampInMilliseconds = Date.now();
 
-    const interval = setInterval(function() {
-      if (startTimestampInMiliseconds + maximumIntervallTimeInMiliseconds <= Date.now()) {
+    var interval = setInterval(function() {
+      if (startTimestampInMilliseconds + maximumIntervalTimeInMilliseconds <= Date.now()) {
```

Comment:
I would prefer to have this as
`if (Date.now() >= startTimestampInMilliseconds + maximumIntervalTimeInMilliseconds) {`
which does exactly the same, but is easier to understand for me personally.
(The way it's written atm really confused me. 😄)